### ISSUE TITLE
When OUTPUT_LANGUAGE is Chinese, generated html file's encoding is GB2312 instead of UTF-8

### ISSUE THREAD

**Issue by zchrissirhcz** [doxygen/doxygen#10086]
**Describe the bug**
I set `OUTPUT_LANGUAGE = Chinese` in Doxyfile, and the generated html file, opened with Browser, shows garbage encoding.

**Expected behavior**

The browser showed html file, should correctly show as Chinese chars.

**Screenshots**

The webpage:
![image](https://github.com/doxygen/doxygen/assets/3831847/1d0fbe32-8534-4396-9803-727e4456523b)

The source code of generated `index.html`, with VSCode:

![image](https://github.com/doxygen/doxygen/assets/3831847/3760cae0-ea71-49af-a5ba-626fc107afb2)

Reopen this index.html, with encoding GB2312:

![image](https://github.com/doxygen/doxygen/assets/3831847/5dcbdcc5-e7d1-4cec-a454-d5feef9f3b07)

Now it correctly show Chinese chars:

![image](https://github.com/doxygen/doxygen/assets/3831847/eb2edbb3-4ba5-4fc4-b5e6-5282fa0575ef)

**To Reproduce**
Attach a self contained example that allows us to reproduce the problem.
Such an example typically exist of some source code (can be dummy code) and a doxygen configuration file used (you can strip it using `doxygen -s -u`). After you verified the example demonstrates the problem, put it in a zip (or tarball) and attach it to the bug report. Try to avoid linking to external sources, since they might disappear in the future.

Doxyfile:
```
PROJECT_NAME = "我的项目"
INPUT = D:/dbg/doxygen_inline_namespace
EXTRACT_ALL = YES
HTML_TIMESTAMP = YES
RECURSIVE = YES
STRIP_FROM_PATH = ../
HTML_OUTPUT=D:/dbg/doxygen_inline_namespace/build/vs2022-x64/html
GENERATE_TREEVIEW = False
HIDE_INLINE_NAMESPACES = False
OUTPUT_LANGUAGE = Chinese
```

hello.h:
```cpp
class Size
{
public:
  int width;
  int height;
};

class Point
{
public:
  float x;
  float y;
};

//namespace {
  namespace foo {

    /// @brief the official API
    struct Image
    {

    };

    inline namespace _v1 {
      
      /// @brief for compatability
      struct Image { };

      struct Tensor { };

      struct Bar
      {

      };

      namespace Yoho {
        struct Yahoo {};
      }
    }

    inline namespace _v2 {

      struct Music
      {

      };

    }

  }

  // class What
  // {
  //   int a;
  // };
//}

namespace rock {
  /// @brief sss
  struct Entity
  {};
}
```

src/057.h:
```
// objective: test inline namespaces
// check: namespacelibrary.xml
// check: namespacelibrary_1_1v1.xml
// check: namespacelibrary_1_1v2.xml

/// the main namespace
namespace library
{
    /// the first namespace
    namespace v1
    {
        /// the class
        class foo {
          public:
            /// member of the class
            void member();
        };
        /// a method
        void func();
        /// a namespace
        namespace NS {}
    }
    /// the inline namespace
    inline namespace v2
    {
        /// v2 the class
        class foo {
          public:
            /// member of the class
            void member();
        };
        /// v2 a method
        void func();
        /// v2 a namespace
        namespace NS {}
    }
}
```

**Version**

Doxygen: 1.9.8 (2f5f67c8a64bc2abad055a03b1b0588b6fa89848*)
System: Windows 11  (`ver` command shows `Microsoft Windows [版本 10.0.22621.1413]`)

**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

**Additional context**

**Comment by albert-github** [comment]
When I run the given example I get:
![image](https://github.com/doxygen/doxygen/assets/5223533/7d53f752-3677-45ef-9cb4-40b7458e0d04)


(I get 2 warnings regarding the input file:
```
```
but they shouldn't influence the issue and I also removed the `INPUT = D:/dbg/doxygen_inline_namespace` and `HTML_OUTPUT=D:/dbg/doxygen_inline_namespace/build/vs2022-x64/html`)

So this looks more like a local problem than to a doxygen problem.

**Comment by zchrissirhcz** [comment]
@albert-github What OS do you use? I previously generated document in Linux (ubuntu), it generate correctly.

**Comment by zchrissirhcz** [comment]
> So this looks more like a local problem than to a doxygen problem.
So what is the proper method to solve this garbage encoding problem? Should I change the OS systemwide encoding..?

**Comment by albert-github** [comment]
I;m not a system administrator and for Windows it is even harder to tell, but probably it might work.

**Comment by zchrissirhcz** [comment]
Solved by https://github.com/doxygen/doxygen/pull/10095

**Comment by albert-github** [comment]
please don't close the issue as this will be done at the moment of an official release

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
